From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Rose <jeremya@chromium.org>
Date: Mon, 2 Aug 2021 15:56:56 -0700
Subject: only create a console if logging to stderr

This fixes an issue on Windows where, when `--enable-logging=file` is
passed and the app was not run from the console, a console window would
be created for each child process, despite logs being redirected to a
file.

diff --git a/content/app/content_main.cc b/content/app/content_main.cc
index 91feb21f4249e103fd84948b34f254a450b28e8c..da2fb0ce9c4a9297db29b800a89399a078d36470 100644
--- a/content/app/content_main.cc
+++ b/content/app/content_main.cc
@@ -305,13 +305,13 @@ RunContentProcess(ContentMainParams params,
 
 #if BUILDFLAG(IS_WIN)
     base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-    if (command_line->HasSwitch(switches::kHeadless)) {
-      // When running in headless mode we want stdio routed however if
-      // console does not exist we should not create one.
-      base::RouteStdioToConsole(/*create_console_if_not_found*/ false);
-    } else if (command_line->HasSwitch(switches::kEnableLogging)) {
-      // Route stdio to parent console (if any) or create one.
-      base::RouteStdioToConsole(/*create_console_if_not_found*/ true);
+    // Route stdio to parent console (if any) or create one.
+    bool const log_to_stderr =
+        command_line->HasSwitch(switches::kEnableLogging) &&
+        command_line->GetSwitchValueASCII(switches::kEnableLogging) != "file" &&
+        !command_line->HasSwitch(switches::kLogFile);
+    if (log_to_stderr) {
+      base::RouteStdioToConsole(true);
     }
 #endif
 
